### ISSUE TITLE
Create a document with all the mixins, functions and variables available in the Portal and Clay and how to use them

### ISSUE THREAD

**Issue by eduardoallegrini** [liferay/liferay-frontend-guidelines#80]
One of the problem I encounter during my latest tasks was the lacking (or apparent absence) of a document where to find mixins, variables and functions altogether.

It could be useful to all the developers, to have a Document (in list or table format) where we can gather these values with a short description or instruction about its use.

Of course we can search through the Clay and Portal files but sometimes there are deprecated results that be confusing (compass, bourbon, bs3_compat_layer, etc)

We should also add the simple information about the imported-clay-files
example:
`@import atlas-variables.scss;`
"what does it means?"
"how can I use it?"
"why I should use it instead of `base-variables` or `clay-atlas`?"

@marcoscv-work, @pat270, @jbalsas What do you think?

**Comment by eduardoallegrini** [comment]
Eventually we should include this https://sass-lang.com/documentation/functions

**Comment by matuzalemsteles** [comment]
I'm afraid this is a bit costly to maintain, @pat270 is using [sassdoc](http://sassdoc.com/) to generate these documents in Clay, we are thinking of adding this to the site.

**Comment by pat270** [comment]
I agree with @matuzalemsteles. I feel like it will get out of date quickly. Regardless, the documentation should be focused on DXP related parts instead of Sass + Compass + Bourbon + Bootstrap + Clay CSS + DXP.

**Comment by jbalsas** [comment]
I think sassdocs looks quite nice and useful @pat270! If we could also bring in those coming from bootstrap it'd be great.

There's one thing that we might want to think about. In my opinion, autogenerated docs are great for reference, and because they're easy to maintain and keep updated. They, however, are not curated at all and lack a clear intent and messaging.

That is, providing reference docs is like doing a brain-dump over our developers. Many of the mixins and variables are mostly for internal consumption. I feel we should make a curated list of what mixins we consider useful and common practice so we can promote their proper usage and explain how they're good to solve specific issues.

**Comment by pat270** [comment]
Thanks @jbalsas! I can document the Bootstrap stuff, just need to restructure the docs a bit. I agree with having a curated list, but I question whether it should all live in this repo. I think best practices related to Clay should live in the Clay repo, that way it's less likely to get out of sync.

**Comment by jbalsas** [comment]
Yeah, agreed, those should live in Clay. From here we can simply link to there
